File: programmers/Graph/가장먼노드.py
from collections import deque

def solution(n, edge):
    answer = 0
    adj=[[] for i in range(n+1)]
    visited=[0]*(n+1)
    for x,y in edge:
        adj[x].append(y)
        adj[y].append(x)
    q=deque()
    q.append(1)
    visited[1]=1
    while q:
        answer=len(q)
        for i in range(len(q)):
            x = q.popleft()
            for nx in adj[x]:
                if visited[nx]==0:
                    visited[nx]=1
                    q.append(nx)   
    return answer

File: javascript/programmers/Graph/가장먼노드.js
function solution(n, edge) {
  var answer = 0;
  let q = [1];
  let adj = Array(n + 1)
    .fill(null)
    .map(() => Array());
  let visited = Array(n + 1).fill(false);
  for (let i of edge) {
    adj[i[0]].push(i[1]);
    adj[i[1]].push(i[0]);
  }
  visited[1] = true;
  while (q.length > 0) {
    answer = q.length;
    let cnt = q.length;
    for (let i = 0; i < cnt; i++) {
      let x = q.shift();
      for (let j = 0; j < adj[x].length; j++) {
        let nx = adj[x][j];
        if (visited[nx] == false) {
          visited[nx] = true;
          q.push(nx);
        }
      }
    }
  }
  return answer;
}
